saleor/plugins/manager.py

```diff
@@ -834,6 +835,45 @@ def list_payment_gateways(
             )
         return gateways
 
+    def list_shipping_methods_for_checkout(
+        self,
+        checkout: "Checkout",
+        channel_slug: Optional[str] = None,
+        active_only: bool = True,
+    ) -> List["ShippingMethodData"]:
+        channel_slug = checkout.channel.slug if checkout else channel_slug
+        plugins = self.get_plugins(channel_slug=channel_slug, active_only=active_only)
+        shipping_plugins = [
+            plugin
+            for plugin in plugins
+            if "get_shipping_methods_for_checkout" in type(plugin).__dict__
```

Comment:
Does the `hasattr` not fit here?

Reply:
Also, make sure that this will work with the latest changes which we introduced to BasePlugin
https://github.com/mirumee/saleor/blob/master/saleor/plugins/base_plugin.py